### PR TITLE
Initial implementation of Exec.

### LINKED ISSUE #9: Support exec in Pod

### PR DESCRIPTION
**Pull request by brendandburns**
Depends on #19 

Fixes #9

### CODE REVIEW

**Comment by brendandburns** [comment]
@mbohlool Now that websockets support is merged, I rebased this on top of that PR.

Please take a look.

Thanks!

**Comment by brendandburns** [comment]
@mbohlool friendly ping on this if you have any feedback.

Thanks!

File: examples/pom.xml [hunk 1]
```diff
@@ -19,6 +19,13 @@
 			<artifactId>client-java-util</artifactId>
 			<version>1.0-SNAPSHOT</version>
 		</dependency>
+		                <!-- https://mvnrepository.com/artifact/com.google.guava/guava -->
```
Comment (mbohlool) [inline examples/pom.xml]: indentation?

File: util/src/main/java/io/kubernetes/client/Exec.java [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+            }
+            return statusCode;
+        }
+  
+        public int exitValue() {
+            return statusCode;
+        }
+
+        public void destroy() {
+            // TODO implementation here
```
Comment (mbohlool) [inline util/src/main/java/io/kubernetes/client/Exec.java]: unnecessary comment?

File: util/src/main/java/io/kubernetes/client/util/Config.java [hunk 3]
```diff
@@ -26,6 +26,8 @@
 
 import javax.net.ssl.KeyManager;
 
+import org.apache.commons.codec.binary.Base64;
```
Comment (mbohlool) [inline util/src/main/java/io/kubernetes/client/util/Config.java]: unused import. is there a way to detect these automatically in a github check? like a linter?

File: util/src/main/java/io/kubernetes/client/util/WebSocketCall.java [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+import java.util.Collections;
+import java.util.Random;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.ThreadPoolExecutor;
+import okio.ByteString;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public final class WebSocketCall {
```
Comment (mbohlool) [inline util/src/main/java/io/kubernetes/client/util/WebSocketCall.java]: where did you use this? I see one WebSocketCall in WebSockets.java but it is not using this class. it uses WebSocketCall in another package.

File: util/src/main/java/io/kubernetes/client/util/WebSockets.java [hunk 5]
```diff
@@ -88,16 +94,25 @@ public static void stream(String path, String method, ApiClient client, SocketLi
...
 
         public Listener(SocketListener listener) {
             this.listener = listener;
         }
 
         @Override
-        public void onOpen(WebSocket webSocket, Response response) {
-            listener.open();
+        public void onOpen(final WebSocket webSocket, Response response) {
+            listener.open("todo", new Closeable() {
```
Comment (mbohlool) [inline util/src/main/java/io/kubernetes/client/util/WebSockets.java]: todo?

File: examples/pom.xml (hunk 1 above)
Comment (brendandburns) [inline examples/pom.xml]: fixed.

File: util/src/main/java/io/kubernetes/client/Exec.java (hunk 2 above)
Comment (brendandburns) [inline util/src/main/java/io/kubernetes/client/Exec.java]: Implemented :)

File: util/src/main/java/io/kubernetes/client/util/Config.java (hunk 3 above)
Comment (brendandburns) [inline util/src/main/java/io/kubernetes/client/util/Config.java]: I can't find a good one. Eclipse will do it, but then I'd have to use eclipse :)

File: util/src/main/java/io/kubernetes/client/util/WebSocketCall.java (hunk 4 above)
Comment (brendandburns) [inline util/src/main/java/io/kubernetes/client/util/WebSocketCall.java]: Removed. (it was part of the GET vs. POST discussion)

File: util/src/main/java/io/kubernetes/client/util/WebSockets.java (hunk 5 above)
Comment (brendandburns) [inline util/src/main/java/io/kubernetes/client/util/WebSockets.java]: Implemented. thanks for the reminder.

**Comment by brendandburns** [comment]
Comments addressed, please re-check.

Thanks!
--brendan

File: examples/pom.xml (hunk 1 above)
Comment (mbohlool) [inline examples/pom.xml]: I can still see the indentation problem here.

**Comment by brendandburns** [comment]
@mbohlool oops, sorry, I formatted the `./util/pom.xml` but missed the other one.  Fixed now, ptal.

Thanks!

**Pull request merged**